src/state_machine.zig

```diff
@@ -339,62 +333,124 @@ pub const StateMachine = struct {
     fn create_transfer(self: *StateMachine, t: Transfer) CreateTransferResult {
         assert(t.timestamp > self.commit_timestamp);
 
-        if (t.flags.padding != 0) return .reserved_flag_padding;
-        if (t.flags.two_phase_commit) {
-            // Otherwise reserved amounts may never be released:
-            if (t.timeout == 0) return .two_phase_commit_must_timeout;
-        } else if (t.timeout != 0) {
-            return .timeout_reserved_for_two_phase_commit;
-        }
-        if (!t.flags.condition and !zeroed_32_bytes(t.reserved)) return .reserved_field;
+        // Either a 2-phase transfer post/void
+        if (t.flags.post_pending_transfer and t.flags.void_pending_transfer) {
+            return .cannot_post_and_void_pending_transfer;
+        } else if (t.flags.post_pending_transfer or t.flags.void_pending_transfer) {
+            if (!t.flags.hashlock and !zeroed_32_bytes(t.reserved)) return .reserved_field;
+            if (t.flags.padding != 0) return .reserved_flag_padding;
 
-        if (t.amount == 0) return .amount_is_zero;
+            const lookup = self.get_transfer(t.id) orelse return .transfer_not_found;
```

Comment:
The motivation for using `user_data` in this way, is so that all pending/posted transfers are then immutable, with their own IDs and with their own timestamps.

Reply:
Resolved.